src/Kiota.Builder/Writers/Python/CodeMethodWriter.cs

```diff
@@ -460,16 +595,82 @@ private void WriteRequestGeneratorBody(CodeMethod codeElement, RequestParams req
+            writer.WriteLine($"writer.{serializationMethodName}(\"{otherProp.WireName}\", self.{otherProp.Name.ToSnakeCase()})");
         }
-        if (additionalDataProperty != null)
-            writer.WriteLine($"writer.write_additional_data_value(self.{additionalDataProperty.Name.ToSnakeCase()})");
     }
+    private void WriteSerializerBodyForUnionModel(CodeClass parentClass, LanguageWriter writer)
+    {
+        var includeElse = false;
+        foreach (var otherProp in parentClass
+                                        .GetPropertiesOfKind(CodePropertyKind.Custom)
+                                        .Where(static x => !x.ExistsInBaseType)
+                                        .OrderBy(static x => x, CodePropertyTypeForwardComparer)
+                                        .ThenBy(static x => x.Name))
+        {
+            writer.StartBlock($"{(includeElse ? "el" : string.Empty)}if self.{otherProp.Name.ToSnakeCase()}:");
+            writer.WriteLine($"writer.{GetSerializationMethodName(otherProp.Type)}(None, self.{otherProp.Name.ToSnakeCase()})");
+            writer.DecreaseIndent();
+            if (!includeElse)
+                includeElse = true;
+        }
+    }
+    private void WriteSerializerBodyForIntersectionModel(CodeClass parentClass, LanguageWriter writer)
+    {
+        var includeElse = false;
+        foreach (var otherProp in parentClass
+                                        .GetPropertiesOfKind(CodePropertyKind.Custom)
+                                        .Where(static x => !x.ExistsInBaseType)
+                                        .Where(static x => x.Type is not CodeType propertyType || propertyType.IsCollection || propertyType.TypeDefinition is not CodeClass)
+                                        .OrderBy(static x => x, CodePropertyTypeBackwardComparer)
+                                        .ThenBy(static x => x.Name))
+        {
+            writer.StartBlock($"{(includeElse ? "el" : string.Empty)}if self.{otherProp.Name.ToSnakeCase()}:");
+            writer.WriteLine($"writer.{GetSerializationMethodName(otherProp.Type)}(None, self.{otherProp.Name.ToSnakeCase()})");
+            writer.DecreaseIndent();
+            if (!includeElse)
+                includeElse = true;
+        }
+        var complexProperties = parentClass.GetPropertiesOfKind(CodePropertyKind.Custom)
+                                            .Where(static x => x.Type is CodeType propType && propType.TypeDefinition is CodeClass && !x.Type.IsCollection)
+                                            .ToArray();
+        if (complexProperties.Any())
+        {
+            if (includeElse)
+            {
+                writer.StartBlock("else:");
+            }
+            var propertiesNames = complexProperties
+                                .Select(static x => x.Name.ToSnakeCase())
+                                .OrderBy(static x => x)
```

Comment:
string comparer (multiple times)

Reply:
Sorry, what's the issue here? It's not clear to me.

Reply:
I believe @baywet is suggesting that we should specify the string comparer when ordering is done. This would also apply to other calls to `OrderBy` introduced  in the PR. 

```suggestion
                                .OrderBy(static x => x, StringComparer.OrdinalIgnoreCase)
```

Reply:
Thanks, got it.